docs-ui/stories/assets/icons/searchPanel.tsx

```diff
@@ -8,10 +8,10 @@ import space from 'sentry/styles/space';
 import {IconData, iconGroups, IconPropName, iconProps, icons} from './data';
 import IconInfoBox from './infoBox';
 
-export type ExtendedIconData = IconData & {
+export interface ExtendedIconData extends IconData {
```

Comment:
Question: are we moving from `type` to `interface` and is this something I should be doing going forward?

Reply:
@malwilley I would not say that we are moving from one to the other as the two are not interchangeable and each have with their own tradeoffs. I would recommend reading the tsc performance doc - I found it to be a great resource https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections when it comes to some of the internals. 

The reason why I changed is related to the type vs interface section described in the ts doc. We have historically had pretty bad compiler performance to a point where ts server would freeze on things like code save actions...

At the time I wrote an audit [doc](https://www.notion.so/sentry/Typescript-performance-audit-ac375cf3d9374e66998a6eab2d5539e4) where I summarized the changes and their impact. The original PR which was later broken down was described [here](https://github.com/getsentry/sentry/pull/30847) with followup PR's that broke it down into smaller chunks https://github.com/getsentry/sentry/pulls?q=is%3Apr+tsc+author%3AJonasBa.

Reply:
Gotcha - I was aware of the performance benefit, but didn't have the other context. Thanks for the thorough explanation!